Gems/Atom/Utils/Code/Source/AssetCollectionAsyncLoader.cpp

```diff
@@ -222,8 +224,8 @@ namespace AZ
 
         Data::AssetBus::MultiHandler::BusConnect(assetId);
         // Kick off asset loading.
-        auto asset = Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::PreLoad);
-        asset.BlockUntilLoadComplete();
+        auto asset = Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::QueueLoad);
+        m_notReadyAssets[assetPath] = asset;
```

Comment:
Should Data::AssetBus::MultiHandler::BusConnect(assetId) be called after the asset is queued and added to the map? Suspect that OnAssetReady might be called twice, when the bus is connected and again after it is queued, if the asset is already loaded.

Reply:
Was thinking about that too. Will change.